OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -81,7 +81,7 @@ public static Target ResolveFrozenActorOrder(this Actor self, Order order, Color
 
 			// Target is still alive - resolve the real order
 			if (frozen.Actor != null && frozen.Actor.IsInWorld)
-				return Target.FromActor(frozen.Actor);
+				return Target.FromFrozenActor(frozen);
```

Comment:
We probably want to do this if the target is dead, too.  We can then drop the workaround below, which causes the "artillery drive up to a dead frozen actor instead of shooting them" bug.